docs/bundler/migration.md

```diff
@@ -0,0 +1,1107 @@
+Bun's bundler API is inspired heavily by [esbuild](https://esbuild.github.io/). Migrating to Bun's bundler from esbuild should be relatively painless. This guide will briefly explain why you might consider migrating to Bun's bundler and provide a side-by-side API comparison reference for those who are already familiar with esbuild's API.
+
+There are a few behavioral differences to note.
+
+- **Bundling by default**. Bun _always bundles by default_. This is why the `--bundle` flag isn't necessary in the Bun example.
+- **It's just a bundler**. Bun's bundler does not include a built-in development server or file watcher. It's just a bundler. The bundler is intended for use in conjunction with `Bun.serve` and other runtime APIs to achieve the same effect. As such, all options relating to HTTP/file watching are not applicable.
+
+## Performance
+
+This is the simplest reason to migrate to Bun's bundler. With an performance-minded API inspired by esbuild coupled with the extensively optimized Zig-based JS/TS parser, Bun's bundler is roughly 50% faster than esbuild on most benchmarks.
+
+IMAGE HERE
+
+## CLI API
+
+Bun and esbuild both provide a command-line interface.
+
+```bash
+$ esbuild <entrypoint> --outdir=out --bundle
+$ bun build <entrypoint> --outdir=out
+```
+
+In Bun's CLI, simple boolean flags like `--minify` do not accept an argument. Other flags like `--outdir <path>` do accept an argument; these flags can be written as `--outdir out` or `--outdir=out`. Some flags like `--define` can be specified several times: `--define foo=bar --define bar=baz`.
+
+{% table %}
+
+- `esbuild`
+- `bun build`
+
+---
+
+- `--bundle`
+- n/a
+- Not necessary, `bun build` always bundles.
+
+---
+
+- `--define:K=V`
+- `--define K=V`
+- Small syntax difference; no colon.
+
+  ```bash
+  $ esbuild --define:K=V
+  $ bun build --define K=V
+  ```
+
+---
+
+- `--external:<pkg>`
+- `--external <pkg>`
+- Small syntax difference; no colon.
+
+---
+
+- `--format`
+- `--format`
+- Bun only supports `"esm"` currently but other module formats are planned. esbuild defaults to `"iife"`.
+
+---
+
+- `--loader`
+- `--loader`
```

Comment:
i think we should match how we explain `--define` above, instead of the full example below we can get away with

- `--loader:.ext=loader`
- `--loader .ext:loader`

and then we should link to the Loaders page for the list of valid loaders, mentioning that `dataurl`, `binary`, `base64`, `copy`, `empty` are not implemented yet.

though jarred might implement these soon. see: https://discord.com/channels/1021987251930005594/1098562907941371914/1101541375419351121